pkg/driver_cryptocell/psa_cc/ecc_common.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (C) 2021 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_psa_crypto pkg_driver_cryptocell
+ * @{
+ *
+ * @brief       Glue code translating between PSA Crypto and the CryptoCell 310 driver APIs
+ *
+ * @author      Lena Boeckmann <lena.boeckmann@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "psa_error.h"
+#include "psa_periph_ecc_common.h"
+#include "cryptocell_util.h"
+
+#define ENABLE_DEBUG    (1)
+#include "debug.h"
+
+extern CRYS_RND_State_t*     rndState_ptr;
+
+CRYS_ECPKI_Domain_t* pDomain;
+SaSiRndGenerateVectWorkFunc_t rndGenerateVectFunc;
+
+psa_status_t periph_common_ecc_generate_key_pair(uint8_t * priv_key_buffer,
+                                                 uint8_t * pub_key_buffer,
+                                                 uint32_t * priv_key_buffer_length,
+                                                 uint32_t * pub_key_buffer_length,
+                                                 CRYS_ECPKI_DomainID_t domain)
+{
+    CRYS_ECPKI_UserPrivKey_t priv_key;
+    CRYS_ECPKI_UserPublKey_t pub_key;
+    CRYS_ECPKI_KG_FipsContext_t FipsBuff;
+    CRYS_ECPKI_KG_TempData_t TempECCKGBuff;
+    int ret;
+
+    rndGenerateVectFunc = CRYS_RND_GenerateVector;
+    pDomain = (CRYS_ECPKI_Domain_t*)CRYS_ECPKI_GetEcDomain(domain);
+
+    cryptocell_enable();
+    ret = CRYS_ECPKI_GenKeyPair(rndState_ptr, rndGenerateVectFunc, pDomain, &priv_key, &pub_key, &TempECCKGBuff, &FipsBuff);
+    cryptocell_disable();
+    if (ret != CRYS_OK){
+        DEBUG("CRYS_ECPKI_GenKeyPair failed with 0x%x \n", ret);
+        return CRYS_to_psa_error(ret);
+    }
+
+    ret = CRYS_ECPKI_ExportPrivKey(&priv_key, priv_key_buffer, priv_key_buffer_length);
+    if (ret != CRYS_OK){
+        DEBUG("CRYS_ECPKI_ExportPrivKey failed with 0x%x \n", ret);
+        return CRYS_to_psa_error(ret);
+    }
+
+    ret = CRYS_ECPKI_ExportPublKey(&pub_key, CRYS_EC_PointUncompressed, pub_key_buffer, pub_key_buffer_length);
+    if (ret != CRYS_OK){
+        DEBUG("CRYS_ECPKI_ExportPubKey failed with 0x%x \n", ret);
+        return CRYS_to_psa_error(ret);
+    }
+
+    return PSA_SUCCESS;
+}
+
+psa_status_t periph_common_ecc_sign_hash(const uint8_t * priv_key,
```

Comment:
```suggestion
psa_status_t periph_common_ecc_sign_hash(const uint8_t *priv_key,
```

More instances in the following lines